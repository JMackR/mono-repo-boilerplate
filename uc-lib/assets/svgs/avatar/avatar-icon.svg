<svg width="96" height="96" viewBox="0 0 96 96">
    <defs>
        <circle id="prefix__a" cx="48" cy="48" r="48"/>
    </defs>
    <g fill="none" fill-rule="evenodd">
        <mask id="prefix__b" fill="#fff">
            <use href="#prefix__a"/>
        </mask>
        <use fill="limestone" href="#prefix__a"/>
        <path fill="#000" d="M48.546 25.5c11.193 0 15.827 7.757 17.64 10.619 3.221 5.083 2.978 14.772-2.188 20.75l-.366.427c-4.718 5.56-3.11 5.728-1.805 7.476.685.917 6.105 2.026 11.47 5.741 3.819 2.645 7.598 6.702 9.953 10.243C74.405 90.136 61.799 96 47.809 96c-13.79 0-26.234-5.697-35.059-14.841 2.169-3.596 5.893-8.303 9.679-10.646 6.284-3.888 11.63-5.048 12.835-5.741l.182-.109c1.888-1.163 2.054-2.61-1.713-7.137l-.556-.657c-4.464-5.198-5.661-14.919-2.272-20.75l.31-.525c1.923-3.207 6.76-10.094 17.33-10.094z" mask="url(#prefix__b)"/>
    </g>
</svg>
