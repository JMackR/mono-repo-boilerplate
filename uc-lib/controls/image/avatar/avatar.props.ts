import { RemoteImageSource } from "../remote-image-source"

export interface AvatarProps {
  source?: RemoteImageSource

  size: number

  children?: React.ReactNode

  isAutosDealer?: boolean

  /**
   * Used to locate this view in end-to-end tests.
   */
  testID?: string
}
