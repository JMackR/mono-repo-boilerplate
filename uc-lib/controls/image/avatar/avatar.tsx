import React from "react"
import { AvatarProps } from "./avatar.props"
import { RemoteImage } from "../remote-image"
import { Flex } from "../../layout"
import { AvatarIcon } from "../../../assets"
import { SVG } from "../svg"

export const Avatar = (props: AvatarProps) => {
  const { size, source, children, testID } = props
  const SIZE_TO_RADIUS_RATIO = 2
  const borderRadius = size / SIZE_TO_RADIUS_RATIO

  return (
    <Flex direction="column" width={size} height={size}>
      {source?.uri && source.uri !== "" ? (
        <RemoteImage
          source={source}
          width={size}
          height={size}
          borderRadius={borderRadius}
          aspectRatio={1}
          resizeMode="cover"
          testID={testID || "uc-lib.avatar-image"}
        />
      ) : (
        <SVG
          localSVG={{
            SVG: AvatarIcon.SVG,
            size: { width: size, height: size },
          }}
        />
      )}
      {children}
    </Flex>
  )
}
