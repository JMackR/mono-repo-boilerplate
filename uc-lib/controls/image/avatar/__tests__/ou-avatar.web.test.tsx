import React from "react"
import * as renderer from "react-test-renderer"
import { Avatar } from "../avatar"

describe("Avatar Snapshot Tests", () => {
  const avatarSize = 100
  const photoURL =
    "https://media-exp1.licdn.com/dms/image/C4E03AQHXHzRJLFryWw/profile-displayphoto-shrink_200_200/0?e=1585785600&v=beta&t=36ZZ3awsowdk_mAFr9oTQ5IS9moXDqNP9jRx_qR1hJU"

  test("Avatar renders properly without children", () => {
    const tree = renderer.create(<Avatar size={avatarSize} source={{ uri: photoURL }} />)
    expect(tree).toMatchSnapshot()
  })

  test("Avatar renders properly with children", () => {
    const tree = renderer.create(
      <Avatar size={avatarSize} source={{ uri: photoURL }}>
        <p>Hype!</p>
      </Avatar>,
    )
    expect(tree).toMatchSnapshot()
  })
})
