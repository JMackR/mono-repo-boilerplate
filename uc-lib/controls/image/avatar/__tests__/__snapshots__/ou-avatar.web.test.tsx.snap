// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snapshot Tests Avatar renders properly with children 1`] = `
.ouflex {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: column;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100px;
  height: 100px;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.container {
  position: relative;
  justify-content: center;
  align-items: center;
  display: block;
  width: 100px;
  height: 0px;
  border-radius: 50px;
  padding-top: 100%;
  overflow: hidden;
  min-width: 100px;
}

.image {
  position: absolute;
  max-width: 100%;
  top: 0px;
  bottom: 0px;
  height: 100%;
  width: 100%;
  object-fit: cover;
  display: block;
}

.children {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 2;
}

<div
  className="ouflex"
>
  <picture
    className="container"
  >
    <source
      srcSet="https://media-exp1.licdn.com/dms/image/C4E03AQHXHzRJLFryWw/profile-displayphoto-shrink_200_200/0?e=1585785600&v=beta&t=36ZZ3awsowdk_mAFr9oTQ5IS9moXDqNP9jRx_qR1hJU"
    />
    <img
      className="image"
      onLoad={[Function]}
      src="https://media-exp1.licdn.com/dms/image/C4E03AQHXHzRJLFryWw/profile-displayphoto-shrink_200_200/0?e=1585785600&v=beta&t=36ZZ3awsowdk_mAFr9oTQ5IS9moXDqNP9jRx_qR1hJU"
    />
    <div
      className="children"
    />
  </picture>
  <p>
    Hype!
  </p>
</div>
`;

exports[`Avatar Snapshot Tests Avatar renders properly without children 1`] = `
.ouflex {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: column;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 100px;
  height: 100px;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.container {
  position: relative;
  justify-content: center;
  align-items: center;
  display: block;
  width: 100px;
  height: 0px;
  border-radius: 50px;
  padding-top: 100%;
  overflow: hidden;
  min-width: 100px;
}

.image {
  position: absolute;
  max-width: 100%;
  top: 0px;
  bottom: 0px;
  height: 100%;
  width: 100%;
  object-fit: cover;
  display: block;
}

.children {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 2;
}

<div
  className="ouflex"
>
  <picture
    className="container"
  >
    <source
      srcSet="https://media-exp1.licdn.com/dms/image/C4E03AQHXHzRJLFryWw/profile-displayphoto-shrink_200_200/0?e=1585785600&v=beta&t=36ZZ3awsowdk_mAFr9oTQ5IS9moXDqNP9jRx_qR1hJU"
    />
    <img
      className="image"
      onLoad={[Function]}
      src="https://media-exp1.licdn.com/dms/image/C4E03AQHXHzRJLFryWw/profile-displayphoto-shrink_200_200/0?e=1585785600&v=beta&t=36ZZ3awsowdk_mAFr9oTQ5IS9moXDqNP9jRx_qR1hJU"
    />
    <div
      className="children"
    />
  </picture>
</div>
`;
